### DISCUSSION TITLE
EMQX 5.7.1 Opensource cluster configuration and feature clarity

### DISCUSSION THREAD

**Discussion by alexchow94** [emqx/emqx#13582] [Q&A] [answered]
Hello,

I was going through this documentation (https://docs.emqx.com/en/emqx/latest/getting-started/feature-comparison.html), which mentioned that OpenSource version of EMQX supports only 3 nodes cluster and upto 100K connections.
Having below queries regarding the same,
1. Is it true that cluster with more than 3 nodes is not possible in opensource version? Because I was able create a 4 nodes cluster using 5.7.1.
2. 100K connections limit is for the entire cluster or per node? And how this limit works in opensource version? since there is a parameter 'max_connections' which is configurable in the listener block that I can set to more than 100K, right?

Thank you for your time and support.

**Comment by keynslug** [comment, marked as answer]
Hi @alexchow94,

Take those numbers as a quick estimation rather than a hard limit. In general you probably can have more than that. Large clusters comprised of core nodes could have questionable stability and availability because of how internal database works. And the number of connections is limited by available computer resources in the first place.

**Reply by alexchow94** [reply]
Thank you @keynslug for the prompt response.

So, in general, I would be able to achieve a 4-node cluster using opensource version 5.7.1 to manage 500K+ connections? (Considering corresponding compute resources are provisioned).

**Reply by id** [reply]
Hi @alexchow94. That is correct, there is no hard coded limit on max number of connections in the opensource version.
